keras_contrib/layers/crf.py

```diff
@@ -282,63 +289,19 @@ def get_config(self):
 
     @property
     def loss_function(self):
-        if self.learn_mode == 'join':
-            def loss(y_true, y_pred):
-                assert self._inbound_nodes, 'CRF has not connected to any layer.'
-                assert not self._outbound_nodes, 'When learn_model="join", CRF must be the last layer.'
-                if self.sparse_target:
-                    y_true = K.one_hot(K.cast(y_true[:, :, 0], 'int32'), self.units)
-                X = self._inbound_nodes[0].input_tensors[0]
-                mask = self._inbound_nodes[0].input_masks[0]
-                nloglik = self.get_negative_log_likelihood(y_true, X, mask)
-                return nloglik
-            return loss
-        else:
-            if self.sparse_target:
-                return sparse_categorical_crossentropy
-            else:
-                return categorical_crossentropy
+        raise AttributeError('"CRF.loss_function" is depreciated. Please use "losses.crf_loss".')
```

Comment:
Instead of raising an Error, is it possible to determine the appropriate loss and supply it here? If not, then this is fine.